js/ui/shellEntry.js

```diff
@@ -184,6 +179,28 @@ var OverviewEntry = GObject.registerClass({
         global.stage.connect('notify::key-focus', this._onStageKeyFocusChanged.bind(this));
     }
 
+    _updateSearchEntryText() {
+
+        let entryText;
+
+        if (this._parentalControlsManager.isContentTypeAllowed("x-scheme-handler/http")) {
+            // Set the search entry's text based on the current search engine
+            let searchEngine = Util.getSearchEngineName();
+
+            if (searchEngine != null)
+                entryText = _("Search %s and more…").format(searchEngine);
+            else
+                entryText = _("Search the internet and more…");
+        } else {
+            entryText = _("Search…");
+        };
+
+        let hintActor = new St.Label({ text: entryText,
+                                       style_class: 'search-entry-text-hint' });
+
+        this.hint_actor = hintActor;
```

Comment:
Why replace the hint actor rather than calling `.set_text()` on the existing one?

Reply:
Oh yea! that's right. Maybe I should spend some time reading documentation/API on `St` .
Re-iterated the PR to use this approach.